python/dolfinx/fem/element.py

```diff
@@ -46,6 +46,56 @@ def dtype(self) -> np.dtype:
         """Scalar type for the coordinate element."""
         return np.dtype(self._cpp_object.dtype)
 
+    @property
+    def dim(self) -> int:
+        """The dimension of the coordinate element space.
+
+        The number of basis function is returned. E.g., for a linear
+        triangle cell the dimension will be 3.
+        """
+        return self._cpp_object.dim
+
+    def create_dof_layout(self) -> _cpp.fem.ElementDofLayout:
+        """Compute and return the dof layout"""
+        return self._cpp_object.create_dof_layout()
+
+    def push_forward(self, X: np.ndarray, cell_geometry: np.ndarray) -> np.ndarray:
```

Comment:
`npt` for typing?

Reply:
Would be a bit nasty, as the typing would be `npt.NDArray[np.float32] | npt.NDArray[np.float64` as we can have either precision. What do you think?

Reply:
I think it's ok.